gazebo_grasp_plugin/src/GazeboGraspGripper.cpp

```diff
@@ -143,16 +156,32 @@ bool GazeboGraspGripper::HandleAttach(const std::string& objName)
     gazebo::math::Pose diff = obj->GetLink()->GetWorldPose() - this->palmLink->GetWorldPose();
     this->palmLink->AttachStaticModel(obj,diff);
 #else
+
+
+#if GAZEBO_MAJOR_VERSION >= 8
+    physics::CollisionPtr obj = boost::dynamic_pointer_cast<physics::Collision>(world->EntityByName(objName));
+#else
     physics::CollisionPtr obj = boost::dynamic_pointer_cast<physics::Collision>(world->GetEntity(objName));
+#endif /* GAZEBO_MAJOR_VERSION */
     if (!obj.get()){
         std::cerr<<"ERROR: Object "<<objName<<" not found in world, can't attach it"<<std::endl;
         return false;
     }
-    gazebo::math::Pose diff = obj->GetLink()->GetWorldPose() - this->palmLink->GetWorldPose();
+
+#if GAZEBO_MAJOR_VERSION >= 8
+    gz_math_Pose3d diff = obj->GetLink()->WorldPose() - this->palmLink->WorldPose();
+    this->fixedJoint->Load(this->palmLink,obj->GetLink(), diff);
+    this->fixedJoint->Init();
+    this->fixedJoint->SetUpperLimit(0, 0);
+    this->fixedJoint->SetLowerLimit(0, 0);
+#else
+    gz_math_Pose3d diff = obj->GetLink()->GetWorldPose() - this->palmLink->GetWorldPose();
```

Comment:
When using a getter for the world pose, then at this point only the `SetUpperLimit()` and `SetLowerLimit()` lines have to be within the #if  (see also my other related review comment)